doc/index.rst

```diff
@@ -1,9 +1,49 @@
 IPv8 Documentation
 ==================
 
-Welcome to the IPv8 documentation!
-This file will help you navigate the existing documentation.
+Welcome to the IPv8 documentation! This file will help you navigate the existing documentation.
+The documentation will explain the concepts behind IPv8, the architecture, and guide you through making your first overlay.
 
+
+Why does IPv8 exist?
+====================
+
+Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our endpoints are not dependent on any central infrastructure. Our architecture is academically pure and fully decentralized; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlays are permissionless and requires no server upkeep costs. 
+
+IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone.
+
+
+IPv8 Features
+=============
+
+IPv8 is a networking layer which offers identities, communication with some robustness, and provides hooks for higher layers. For instance, our Internet-deployed reputation functions and ledger-based storage of reputation data. In combination with our Trustchain, IPv8 is designed as a mechanism to build trust.  Each network overlay offers network connections to known digital identities through public keys. Overlays are robust against several network problems and security issues. Using a custom NAT-traversing DHT to find the current IPv4 network address, IPv8 keeps the network connectivity going even as the IPv4 addresses change.  Each network overlay keeps track of a number of neighbors and occasionally checks if they are still responsive.
+
+IPv8 offers global connectivity through integrated UDP NAT puncturing, announcement of your identity claim and a web-of-trust. IPv8 has an integrated attestation service. You can use IPv8 for official verification that something is true or authentic, according to a trustworthy attestor. By using zero-knowledge proofs we attempt to minimize privacy leakage.
```

Comment:
Suggestion: we talk a lot about components here. We could add links to the relevant directories implementing this component?

Reply:
I don't have readthedocs documenting the actual source code, so I can't really add simple links to the appropriate classes.